components/sidebar/sidebar_category_list/index.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch} from 'redux';
+import {createSelector} from 'reselect';
+
+import {getSortedUnreadChannelIds, getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
+import {makeGetCategoriesForTeam, makeGetChannelsForCategory} from 'mattermost-redux/selectors/entities/channel_categories';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+import {GenericAction} from 'mattermost-redux/types/actions';
+import {ChannelCategory} from 'mattermost-redux/types/channel_categories';
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {switchToChannelById} from 'actions/views/channel';
+import {close} from 'actions/views/lhs';
+import {isUnreadFilterEnabled} from 'selectors/views/channel_sidebar';
+import {GlobalState} from 'types/store';
+
+import SidebarCategoryList from './sidebar_category_list';
+
+const getCategoriesForTeam = makeGetCategoriesForTeam();
+const getChannelsForCategory = makeGetChannelsForCategory();
```

Comment:
Is this intentional? This will effectively have only one instance of these functions and if we are using them for multiple components then memoization does not work.